### PR TITLE
Lock down permissions for access to performance tracking

### PR DESCRIPTION
**Pull request by arthurashman**
## Description of change
Ensure that should a user_manager have a Supervisor role in prod that they cannot see the Performance tracking nav link

Make user_manager permissions in staging when holding a caseworker role the same as they would be for a non-user manager caseworker in prod for QA and debugging.

## Notes for reviewer
Please can someone read the performance_tracking/viewing_spec fairly carefully to check my logic.

My assumption has been to that if manually testing a caseworker-only feature on staging a dev will still be set as a user_manager and will want to see what a caseworker would see in prod. And if manually testing a supervisor feature, then would want to see what a supervisor can see and therefore will need to get another user_manager to toggle them to be a supervisor - and only then see the feature.

### CODE REVIEW

**Review by timpeat — COMMENTED** [review]
I'm wondering if I shouldn't bring forward the per user reports work. Which, I think, will remove quite a bit of the pain. I'll can try to pull together a spike tomorrow.

File: app/models/user.rb [hunk 1]
```diff
@@ -51,6 +51,15 @@ def name_or_email
     email
   end
 
+  def service_user_supervisor?
+    FeatureFlags.basic_user_roles.enabled? && supervisor? && !can_manage_others?
+  end
+
+  # A dev only user state for debugging in staging, not found in production environments
+  def admin_user_supervisor?
+    FeatureFlags.basic_user_roles.enabled? && supervisor? && can_manage_others?
```
Comment (hiboabd) [inline app/models/user.rb:60]: If a supervisor can already see the performance tracking tab on all environments then I don't see the point of the  `can_manage_others` check. By enabling it for user managers on staging we have to be referring to user managers with the caseworker role

File: spec/system/header_nav_spec.rb [hunk 2]
```diff
@@ -81,10 +81,18 @@
...
 
-      it 'shows the "Performance tracking" link and it can be followed' do
-        expect { click_link('Performance tracking') }.to change {
-          page.first('.govuk-heading-xl').text
-        }.from('Your list').to('Performance tracking')
+      it 'does not show the "Performance tracking" link' do
+        expect(page).not_to have_link('Performance tracking')
+      end
+
+      context 'when user managers are logged in as supervisors' do
```
Comment (mseedat-moj) [inline spec/system/header_nav_spec.rb:88]: This test makes it sounds like the user manager can change their role perhaps 'when user manager is supervisor' ?

File: spec/system/performance_tracking/viewing_spec.rb [hunk 3]
```diff
@@ -13,16 +13,49 @@
         heading_text = page.first('.govuk-heading-xl').text
         expect(heading_text).to eq('Manage users')
       end
+
+      it 'does not show the performance tracking nav link' do
+        expect(page).not_to have_link('Performance tracking')
+      end
+
+      context 'when user manager role is supervisor but without service access enabled' do
```
Comment (mseedat-moj) [inline spec/system/performance_tracking/viewing_spec.rb:21]: I think this test should explicitly say it is testing the feature flag rather general than system behaviour, maybe even explicitly set the feature flag value to `false`

File: spec/system/performance_tracking/viewing_spec.rb [hunk 4]
```diff
@@ -51,26 +84,22 @@
...
-
-      context 'when logged in as a user manager' do
-        let(:current_user_can_manage_others) { true }
-
-        it 'can access "Performance tracking" page as a user manager' do
-          heading_text = page.first('.govuk-heading-xl').text
-          expect(heading_text).to eq('Performance tracking')
-        end
+        expect(heading_text).to eq('Performance tracking')
       end
```
Comment (mseedat-moj) [inline spec/system/performance_tracking/viewing_spec.rb:103]: It seems like there is a whole bunch of tests dedicated to testing the same behaviours with/without the feature flag enabled so maybe laying out the tests with a describe block for 'with feature flag' and 'without feature flag' and then a shared example that runs in both with different expected results will make it a bit clearer?

File: app/models/user.rb (hunk 1 above)
Comment (arthurashman) [reply app/models/user.rb:60]: It pairs with the `service_user_manager?` method that checks that the user is not a user manager.

Those who can see the tab are:
Service supervisors
Admin supervisors AND service access turned on

Everyone else cannot see the nav link or tab content

Comment (hiboabd) [reply app/models/user.rb:60]: But on staging what would be the point of setting yourself to be a user manager if you're already a supervisor and therefore should be able to see the page?

File: spec/system/performance_tracking/viewing_spec.rb (hunk 4 above)
Comment (arthurashman) [reply spec/system/performance_tracking/viewing_spec.rb:103]: I've tidied things up a bit, but left it as When can access vs When cannot access. I'll DM you

**Comment by arthurashman** [comment]
Not the approach we are taking - different user roles will have access to different reports

**Pull request closed without merging**